packages/web3-utils/test/fixtures/converters.ts

```diff
@@ -0,0 +1,241 @@
+	],
+	['0x0c1g', 'Invalid value given "0x0c1g". Error: not valid hex string.'],
+	['0c1g', 'Invalid value given "0c1g". Error: not valid hex string.'],
+	['data', 'Invalid value given "data". Error: not valid hex string.'],
+	[12, 'Invalid value given "12". Error: can not parse as byte data.'],
+	[['string'], 'Invalid value given "string". Error: contains invalid integer values.'],
+	[null, 'Invalid value given "null". Error: can not parse as byte data.'],
+	[undefined, 'Invalid value given "undefined". Error: can not parse as byte data.'],
+	[{}, 'Invalid value given "[object Object]". Error: can not parse as byte data.'],
+];
+
+export const numberToHexValidData: [Numbers, HexString][] = [
+	[1, '0x1'],
+	[255, '0xff'],
+	[256, '0x100'],
+	[54, '0x36'],
+	[BigInt(12), '0xc'],
+	['768', '0x300'],
+	['-768', '-0x300'],
+	[-255, '-0xff'],
+	['0xFF0', '0xff0'],
+	['-0xa0', '-0xa0'],
+	[0xff, '0xff'],
+	[-0xff, '-0xff'],
+];
+
+export const numberToHexInvalidData: [any, string][] = [
+	[12.2, 'Invalid value given "12.2". Error: not a valid integer.'],
+	['0xag', 'Invalid value given "0xag". Error: not a valid integer.'],
+	['122g', 'Invalid value given "122g". Error: not a valid integer.'],
+	[null, 'Invalid value given "null". Error: not a valid integer.'],
+	[undefined, 'Invalid value given "undefined". Error: not a valid integer.'],
+	[{}, 'Invalid value given "[object Object]". Error: not a valid integer.'],
+];
+
+export const hexToNumberValidData: [HexString, Numbers][] = [
+	['0x1', 1],
+	['0xff', 255],
+	['0x100', 256],
+	['0x36', 54],
+	['0xc', 12],
+	['0x300', 768],
+	['-0x300', -768],
+	['-0xff', -255],
+	['0XFF0', 4080],
+	['-0xa0', -160],
+	['0x1fffffffffffff', 9007199254740991], // Number.MAX_SAFE_INTEGER
+	['0x20000000000000', BigInt('9007199254740992')], // Number.MAX_SAFE_INTEGER + 1
+	['-0x1fffffffffffff', -9007199254740991], // Number.MIN_SAFE_INTEGER
+	['-0x20000000000000', BigInt('-9007199254740992')], // Number.MIN_SAFE_INTEGER - 1
+];
+
+export const hexToNumberInvalidData: [HexString, string][] = [
+	['1a', 'Invalid value given "1a". Error: not valid hex string.'],
+	['0xffdg', 'Invalid value given "0xffdg". Error: not valid hex string.'],
+	['xfff', 'Invalid value given "xfff". Error: not valid hex string.'],
+	['-123', 'Invalid value given "-123". Error: not valid hex string.'],
+	['-9x123', 'Invalid value given "-9x123". Error: not valid hex string.'],
+];
+
+export const utf8ToHexValidData: [string, HexString][] = [
+	['I have 100£', '0x49206861766520313030c2a3'],
+	['I \u1234data', '0x4920e188b464617461'],
+	['I ♥ data', '0x4920e299a52064617461'],
+	['I \u0000 data', '0x4920002064617461'],
+	['\u0000 null suffix', '0x206e756c6c20737566666978'],
+	['null prefix\u0000', '0x6e756c6c20707265666978'],
+	['\u0000', '0x'],
+];
+
+export const utf8ToHexInvalidData: [any, string][] = [
+	[12, 'Invalid value given "12". Error: not a valid string.'],
+	[BigInt(12), 'Invalid value given "12". Error: not a valid string.'],
```

Comment:
```suggestion
	[BigInt(12), 'Invalid value given "12". Error: not a valid string.'],
	[12n, 'Invalid value given "12". Error: not a valid string.'],
```

Add test case for alternative syntax for BigInt